platforms/common/include/px4_platform_common/board_common.h

```diff
@@ -258,6 +259,15 @@
 #  define BOARD_EEPROM_WP_CTRL(on_true)
 #endif
 
+/*
+ * Support both RC_SERIAL_PORT and CONFIG_BOARD_SERIAL_RC
+ */
+#if defined(CONFIG_BOARD_SERIAL_RC)
+#  define RC_SERIAL CONFIG_BOARD_SERIAL_RC
+#else
+#  define RC_SERIAL RC_SERIAL_PORT
```

Comment:
Maybe trigger a warning if both `CONFIG_BOARD_SERIAL_RC` and `RC_SERIAL_PORT` are defined?